performance-test/README.md

```diff
@@ -0,0 +1,148 @@
+
+### Execution points for Resident Service endpoints
+
+* The id and access token generated from the Login​ Redirect URI api will be stored in a file and will be used in the headers the api and can be re-used until they are not expired.
+* Test duration is already defined as a variable and needs to be changed as per the test run duration requirement in the user defined variable section.
+* The script is designed to run for 100 users, load being distributed accordingly for each thread group based on the weightage given for each scenario.
+
+
+### Exact steps of execution
+
+	Step 1: Enable only Create Identities in MOSIP Authentication System (Setup) thread group and toggle/disable the remaining thread groups in the script to create the required no of identities.
+	Step 2: Enable only Resident Id Access Token Creation (Setup) thread group and toggle/disable the remaining thread groups in the script to create access tokens and id tokens.
+	Step 3: Enable only Create Resident EventId (Setup) thread group and toggle/disable the remaining thread groups in the script to create the event id's.
+	Step 4: Enable the rest of all the Execution based scenario thread groups and toggle/disable the first 3 setup based thread groups. 
+	Step 5: Make sure test duration and ramp-up is defined in the user defined variable section. 
+	Step 5: Click on Run/Excute the test.
+	Step 6: Monitor the metrics during the test run using reports from jmeter, kibana, grafana and Jprofiler.
+	
+### Designing the workload model for performance test execution
+
+* Calculation of number of users depending on Transactions per second (TPS) provided by client
+
+* Applying little's law
+	* Users = TPS * (SLA of transaction + think time + pacing)
+	* TPS --> Transaction per second.
+
+* For the realistic approach we can keep (Think time + Pacing) = 1 second for API testing
+	* Calculating number of users for 10 TPS
+		* Users= 10 X (SLA of transaction + 1)
+		       = 10 X (1 + 1)
+			   = 20
+			   
+### Usage of Constant Throughput timer to control Hits/sec from JMeter
+
+* In order to control hits/ minute in JMeter, it is better to use Timer called Constant Throughput Timer.
+
+* If we are performing load test with 10TPS as hits / sec in one thread group. Then we need to provide value hits / minute as in Constant Throughput Timer
+	* Value = 10 X 60
+			= 600
+
+* Dropdown option in Constant Throughput Timer
+	* Calculate Throughput based on as = All active threads in current thread group
+		* If we are performing load test with 10TPS as hits / sec in one thread group. Then we need to provide value hits / minute as in Constant Throughput Timer
+	 			Value = 10 X 60
+					  = 600
+		  
+	* Calculate Throughput based on as = this thread
+		* If we are performing scalability testing we need to calculate throughput for 10 TPS as 
+          Value = (10 * 60 )/(Number of users)
+
```

Comment:
Lets add the line, it is preconfigured as per 100 tps, if you are testing other tps, this values to be adjusted.
Also lets tell where it should adjusted

Reply:
Reviewed.